File: client/src/app/modules/miscellaneous/services/guichet-unique/status.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Status } from 'src/app/modules/quotation/model/guichet-unique/referentials/Status';
import { AppRestService } from 'src/app/services/appRest.service';

@Injectable({
  providedIn: 'root'
})
export class StatusService extends AppRestService<Status>{

  constructor(http: HttpClient) {
    super(http, 'quotation/guichet-unique');
  }

  getStatus() {
    return this.getList(new HttpParams(), 'status');
  }

}                        

File: src/main/java/com/jss/osiris/modules/quotation/service/guichetUnique/referentials/StatusService.java
package com.jss.osiris.modules.quotation.service.guichetUnique.referentials;

import java.util.List;

import com.jss.osiris.modules.quotation.model.guichetUnique.referentials.Status;

public interface StatusService {
    public List<Status> getStatus();

    public Status getStatus(String code);
}
